core/batcher.go

```diff
@@ -0,0 +1,348 @@
+	traceCtx, span := tracingHandler.StartNewSpan(traceCtx, BatcherBatchExecute)
+	defer span.End()
+
+	logger := Logger()
+	batchID := stub.GetTxID()
+	span.SetAttributes(attribute.String("batch_tx_id", batchID))
+	start := time.Now()
+	defer func() {
+		logger.Infof("batch %s elapsed time %d ms", batchID, time.Since(start).Milliseconds())
+	}()
+
+	if len(arguments) != 1 {
+		return nil, fmt.Errorf("expected 1 argument, got %d", len(arguments))
+	}
+
+	var batchDTO BatcherBatchExecuteRequestDTO
+	if err := json.Unmarshal([]byte(arguments[0]), &batchDTO); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal BatcherBatchExecuteRequestDTO: %w", err)
+	}
+
+	batcher, err := NewBatcher(stub, cfgBytes, cc, tracingHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create batchInsertHandler: %w", err)
+	}
+
+	if err = batcher.ValidateCreator(creatorSKI, hashedCert); err != nil {
+		return nil, fmt.Errorf("failed to validate creator: %w", err)
+	}
+
+	batchResponse, batchEvent, err := batcher.HandleBatch(traceCtx, batchDTO.Requests)
+	if err != nil {
+		return nil, fmt.Errorf("failed handling batch: %w", err)
+	}
+
+	eventData, err := pb.Marshal(batchEvent)
+	if err != nil {
+		logger.Errorf("Couldn't marshal batch event %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "marshalling batch event failed")
+
+		return nil, fmt.Errorf("failed to marshal batch event: %w", err)
+	}
+	if err = stub.SetEvent(BatcherBatchExecuteEvent, eventData); err != nil {
+		logger.Errorf("Couldn't set batch event %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "set batch event failed")
+
+		return nil, fmt.Errorf("failed set batch event %s: %w", batchID, err)
+	}
+
+	data, err := pb.Marshal(batchResponse)
+	if err != nil {
+		logger.Errorf("Couldn't marshal batch response %s: %s", batchID, err.Error())
+		span.SetStatus(codes.Error, "marshalling batch response failed")
+
+		return nil, fmt.Errorf("failed marshalling batch response %s: %w", batchID, err)
+	}
+
+	return data, nil
+}
+
+const skiPattern = "^[0-9a-f]+$"
+
+func validateBatcherSKI(batcherSKI string) error {
+	ok, err := regexp.MatchString(skiPattern, batcherSKI)
+	if err != nil {
+		return fmt.Errorf("failed to compile regexp pattern ski: %w", err)
+	}
+
+	if !ok {
+		return fmt.Errorf("invalid batcher SKI %s: does not match regex pattern '%s'", batcherSKI, skiPattern)
+	}
+
+	return nil
+}
+
+func NewBatcher(stub shim.ChaincodeStubInterface, cfgBytes []byte, cc *ChainCode, tracingHandler *telemetry.TracingHandler) (Batcher, error) {
+	contractCfg, err := config.ContractConfigFromBytes(cfgBytes)
+	if err != nil {
+		return Batcher{}, fmt.Errorf("LoadConfig: contract config from bytes: %w", err)
+	}
+
+	batchCacheStub := cachestub.NewBatchCacheStub(stub)
+
+	batcherSKI := contractCfg.GetBatcherSKI()
+	err = validateBatcherSKI(batcherSKI)
+	if err != nil {
+		return Batcher{}, fmt.Errorf("failed to validate batcher SKI: %w", err)
+	}
+
+	return Batcher{
```

Comment:
Why is a copy being returned here with return Batcher when all the methods have a pointer receiver?